### ISSUE TITLE
Improvement request: text input should always have focus

### ISSUE THREAD

**Issue by smakkerlak** [squidowl/halloy#446]
Currently, clicking a channel in halloy while another application has focus, or clicking a different channel than the currently focused one, automatically makes the text input active so you can type.

I keep clicking something that moves focus away from the text input and start typing into nothing. Send help.

Basically, I'd love it if any window with a text input would always have its focus be the text input, no matter what the user does. This includes things like clicking empty areas in nicklist or channel bar. I cannot currently think of a situation where you would want focus to not be the text input, expect perhaps if there are multiple text input fields and one was clicked explicitly (not sure if any exists)

Running Win11, built from latest main

**Comment by dmd** [comment]
Yes. If I am in another window, and I click ANYWHERE in halloy, the text input most closely associated with where I click should have focus. There is no situation permitted where hallow should be the frontmost app but no text input has focus.

**Comment by voidlily** [comment]
Agreed, I'm coming at this from muscle memory in hexchat where no matter where you clicked in the window, the text input would always have focus, and the current behavior is throwing me off a lot
